xrootd/xrdfs/fs.go

```diff
@@ -12,4 +12,61 @@ import (
 // FileSystem implements access to a collection of named files over XRootD.
 type FileSystem interface {
 	Dirlist(ctx context.Context, path string) ([]EntryStat, error)
+	Open(ctx context.Context, path string, mode OpenMode, options OpenOptions) (File, error)
 }
+
+// OpenMode is the mode in which path is to be opened.
+// The mode is an "or`d" combination of ModeXyz flags.
+type OpenMode uint16
+
+const (
+	OpenModeOwnerRead    OpenMode = 0x100 // OpenModeOwnerRead indicates that owner has read access.
+	OpenModeOwnerWrite   OpenMode = 0x080 // OpenModeOwnerWrite indicates that owner has write access.
+	OpenModeOwnerExecute OpenMode = 0x040 // OpenModeOwnerExecute indicates that owner has execute access.
+
+	OpenModeGroupRead    OpenMode = 0x020 // OpenModeGroupRead indicates that group has read access.
+	OpenModeGroupWrite   OpenMode = 0x010 // OpenModeGroupWrite indicates that group has write access.
+	OpenModeGroupExecute OpenMode = 0x008 // OpenModeGroupExecute indicates that group has execute access.
+
+	OpenModeOtherRead    OpenMode = 0x004 // OpenModeOtherRead indicates that owner has read access.
+	OpenModeOtherWrite   OpenMode = 0x002 // OpenModeOtherWrite indicates that owner has write access.
+	OpenModeOtherExecute OpenMode = 0x001 // OpenModeOtherExecute indicates that owner has execute access.
+)
+
+// OpenOptions are the options to apply when path is opened.
+type OpenOptions uint16
+
+const (
+	// OpenOptionsNone specifies that file is opened without specific options.
+	OpenOptionsNone OpenOptions = 0
+	// OpenOptionsCompress specifies that file is opened even when compressed.
+	OpenOptionsCompress OpenOptions = 1
```

Comment:
could we replace these with the explicit way of defining a bitmask? (ie: `1<<0`, `1<<1`, ... (or `1<<iota`))